File: examples/thread-python/intro01.py
## =================================================================
##
## File: intro01.cpp
## Author: Pedro Perez
## Description: This file contains an example of threads in Python. 
##
## Copyright (c) 2022 by Tecnologico de Monterrey.
## All Rights Reserved. May be reproduced for any non-commercial
## purpose.
##
## =================================================================

import multiprocessing as mp
import threading as th

# CORES = multiprocessingmp.cpu_count() 
CORES = 4
LIMIT = 3

def task():
    id = th.current_thread().ident
    for i in range(0, LIMIT):
        print("Process id = ", id, " i = " , i)

if __name__ == "__main__":
    t = th.Thread(target=task)
    t.start()
    print("Main    : before joining process 0.")
    t.join()
    print("Main    : process 0 done.")

File: examples/01-intro/intro01.cpp
// =================================================================
//
// File: intro01.cpp
// Author: Pedro Perez
// Description: This file show how to create a simple thread.
//              To compile:
//              g++ -o app -pthread intr01.cpp
//
// Copyright (c) 2024 by Tecnologico de Monterrey.
// All Rights Reserved. May be reproduced for any non-commercial
// purpose.
//
// =================================================================

#include <iostream>
#include <iomanip>
#include <thread>

using namespace std;

void task () {
    for (int i = 1; i <= 10; i++) {
        cout << i << " ";
    }
    cout << "\n";
}

int main(int argc, char* argv[]) {
    thread t;

    t = thread(task);

    t.join();

    return 0;
}
